### PR TITLE
Add `beEmpty` matcher for `SetAlgebra`

### PR DESCRIPTION
**Pull request by ikesyo**
Partially implements #496.

### CODE REVIEW

File: Tests/NimbleTests/Matchers/BeEmptyTest.swift [hunk 1]
```diff
@@ -34,16 +34,12 @@ final class BeEmptyTest: XCTestCase, XCTestCaseProvider {
 
         expect("").to(beEmpty())
         expect("foo").toNot(beEmpty())
+
+        expect([] as NSSortOptions).to(beEmpty())
```
Comment (sharplet) [inline Tests/NimbleTests/Matchers/BeEmptyTest.swift]: It feels slightly weird to rely on `NSSortOptions` here, how do you feel about creating a `TestOptionSet` type for testing purposes?

File: Tests/NimbleTests/Matchers/BeEmptyTest.swift [hunk 2]
```diff
@@ -73,5 +69,35 @@ final class BeEmptyTest: XCTestCase, XCTestCaseProvider {
         failsWithErrorMessage("expected to be empty, got <foo>") {
             expect("foo").to(beEmpty())
         }
+
+        failsWithErrorMessage("expected to not be empty, got <NSSortOptions(rawValue: 0)>") {
```
Comment (sharplet) [inline Tests/NimbleTests/Matchers/BeEmptyTest.swift]: Having our own option set type here would also decouple us from `NSSortOptions`'s implementation of string conversion.

File: Tests/NimbleTests/Matchers/BeEmptyTest.swift (hunk 1 above)
Comment (ikesyo) [reply Tests/NimbleTests/Matchers/BeEmptyTest.swift]: Makes sense :+1:

**Comment by ikesyo** [comment]
@sharplet Addressed your feedback!

**Pull request merged**